lib/open_api_spex/controller.ex

```diff
@@ -0,0 +1,153 @@
+defmodule OpenApiSpex.Controller do
+  @moduledoc ~S'''
+  Generation of OpenAPI documentation via ExDoc documentation and tags.
+
+  ## Supported OpenAPI fields
+
+  Attribute `operationId` is automatically provided by the implementation
+  and cannot be changed in any way. It is constructed as `Module.Name.function_name`
+  in the same way as function references in backtraces.
+
+  ### `description` and `summary`
+
+  Description of endpoint will be filled with documentation string in the same
+  manner as ExDocs, so first line will be used as a `summary` and whole
+  documentation will be used as `description` field.
+
+  ### `parameters`
+
+  Parameters of the endpoint are defined by `:parameters` tag which should be
+  map or keyword list that is formed as:
+
+  ```
+  [
+    param_name: definition
+  ]
+  ```
+
+  Where `definition` is `OpenApiSpex.Parameter.t()` structure or map or keyword
+  list that accepts the same arguments.
+
+  ### `responses`
+
+  Responses are controlled by `:responses` tag and currently supports only
+  `application/json` responses. Responses must be defined as a map or keyword list
```

Comment:
I suspect eventually we'll need to support "application/vnd.api+json" responses, perhaps through a configuration parameter to set the default response content type.

Reply:
It can be easily improved by requiring 3-ary tuple as an value for return value which will also contain content type value.

Reply:
Yeah, we'll need `application/vnd.api+json` right away.

Reply:
Resolved, now `:body` and `:responses` requires 3-ary tuples where 2nd argument is MIME type.